source/_data/glossary.yml

```diff
@@ -15,15 +15,15 @@
 - topic: Customize
   description: "[Customization](/docs/configuration/customizing-devices/) allows you to overwrite the default parameters of your devices in the configuration."
 - topic: Device
-  description: "A device is usually a physical unit which can do or observe something."
+  description: "A device is a named collection of entities that all represent the same physical/logical unit, which can do or observe something. An example for a device would be a smart plug named 'Coffee Machine' which provides a `switch` entity plus one or more `sensor` entities for power monitoring or similar."
 - topic: Device tracker
   description: "[Device trackers](/integrations/device_tracker) are used to track the presence, or location, of a device."
 - topic: Discovery
   description: "[Discovery](/integrations/discovery/) is the automatic setup of zeroconf/mDNS and uPnP devices after they are discovered."
 - topic: Domain
   description: "Entities and services belong to a domain, which is the first part of the entity or service, before the `.`. For example `light.kitchen` is an entity in the `light` domain, while `homeassistant.turn_on` is the `turn_on` service for the `homeassistant` domain."
 - topic: Entity
-  description: "An [entity](/docs/configuration/platform_options/) is the representation of a function of a single device, unit, or web service. There may be multiple entities for a single device, unit, or web service, or there may be only one."
+  description: "Entities are things with a unique ID, a name and some kind of state that may be any primitive type such as a number or a string. As the core building block of state in Home Assistant, an [entity](/docs/configuration/platform_options/) is the representation of a functionality of a single device, unit, or web service. There may be multiple entities for a single device, unit, or web service."
```

Comment:
Hmm right, this description is more from the era when we didn't have the concept of "devices" in Home Assistant. So indeed needs an update.

Some points:

- Entity IDs don't have to have a unique ID, also, that is more an implementation detail I guess.
- I don't think we need to tell about data types (numbers, strings, etc)
- Let's avoid terms like "things" and "some kind of".
- The link to the platform options should be dropped. It doesn't apply to all anymore, and additionally, cannot be applied to most.
- "There may be multiple entities for a single device, unit, or web service." -> We should change this to refer back to the "Device" item.
- "web service" should become "service", as we call it "devices & services" nowadays.